event.c

```diff
@@ -2578,7 +2578,7 @@ evthread_notify_base_default(struct event_base *base)
 #ifdef _WIN32
 	r = send(base->th_notify_fd[1], buf, 1, 0);
 #else
-	r = write(base->th_notify_fd[1], buf, 1);
```

Comment:
Why do you did cast to `int`, instead of converting `r` to `ssize_t`? And more importantly, how this is related to the initial problem? Let's revert this change

Reply:
Highest value possible would be `1`, so no need for ssize_t?

Reply:
This is just general sense, function return `ssize_t` let just use the same type. But anyway this change is not related for the this PR and should be removed.